File: Java/Eta/Source/impl/com/thomsonreuters/proxy/authentication/ProxyAuthenticatorResponse.java
package com.thomsonreuters.proxy.authentication;

public class ProxyAuthenticatorResponse implements IProxyAuthenticatorResponse
{
    private final boolean _isProxyConnectionClose;
    private final String _proxyAuthorization;
	
    protected ProxyAuthenticatorResponse(boolean isProxyConnectionClose, String proxyAuthorization)
    {
        _isProxyConnectionClose = isProxyConnectionClose;
        _proxyAuthorization = (proxyAuthorization != null) ? proxyAuthorization : "";
    }

    @Override
    public boolean isProxyConnectionClose()
    {
        return _isProxyConnectionClose;
    }

    @Override
    public String getProxyAuthorization()
    {
        return _proxyAuthorization;
    }

}

File: CSharp/Eta/Src/Transport/transport/proxy/ProxyAuthenticatorResponse.cs
/*|-----------------------------------------------------------------------------
 *|            This source code is provided under the Apache 2.0 license      --
 *|  and is provided AS IS with no warranty or guarantee of fit for purpose.  --
 *|                See the project's LICENSE.md for details.                  --
 *|           Copyright (C) 2022 Refinitiv. All rights reserved.              --
 *|-----------------------------------------------------------------------------
 */

namespace Refinitiv.Eta.Transports.proxy
{
    internal class ProxyAuthenticatorResponse
    {
        public ProxyAuthenticatorResponse(bool isProxyConnectionClose, string proxyAuthorization)
        {
            IsProxyConnectionClose = isProxyConnectionClose;
            ProxyAuthorization = proxyAuthorization;
        }

        public bool IsProxyConnectionClose { get; private set; }

        public string ProxyAuthorization { get ; private set; } 
    }
}
